src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -84,26 +89,68 @@ export default class SimidMapCreative extends BaseSimidCreative {
    * Prompts the users to grant or deny access to their current location.
    * @private 
   */
-  grantLocationAccess_() {
-    this.loadMap_();
-  }
+ grantLocationAccess_() {
+  this.loadMap_();
+}
 
-  /**
-   * Loads a map object that currently displays a hardcoded location.
-   * @param {!google.maps.LatLng=} coordinates The LatLng object of user's current location.
-   * TODO(kristenmason@): implement grant location access and modify
-   * function to pass in current position (currently coords default to GooglePlex)
-   * @private 
-  */
-  loadMap_(coordinates = new google.maps.LatLng(37.422004,-122.081402)) { 
-    const map = new google.maps.Map(document.getElementById('map'), {
-      zoom: DEFAULT_ZOOM,
-      center: coordinates
-    });
-    const marker = new google.maps.Marker({
-      position: coordinates,
-      map: map,
-      title: 'Current Position'
-    });
-  }
+/**
+ * Loads a map object that currently displays a hardcoded location.
+ * @param {!google.maps.LatLng=} coordinates The LatLng object of user's current location.
+ * @private 
+*/
+loadMap_(coordinates = new google.maps.LatLng(37.422004, -122.081402)) {
+  map = new google.maps.Map(document.getElementById('map'), {
+    zoom: DEFAULT_ZOOM,
+    center: coordinates
+  });
+  const marker = new google.maps.Marker({
+    position: coordinates,
+    map: map,
+    title: 'Current Position'
+  });
+  this.findNearby_(searchQuery, coordinates);
+}
+
+/**
+ * Searches for the closest corresponding businesses based off of the given search parameter,
+ * and places pins on the map that represent the 4 closest locations.
+ * @param {!google.maps.LatLng=} coordinates The LatLng object of user's current location.
+ * Currently defaults to GooglePlex coords, will take in user's current location.
+ * @param {String} searchParameter A string with the business's name to use in the query.
+ * @private 
+*/
+findNearby_(searchParameter, coordinates) {
+  const request = {
+    location: coordinates,
+    name: searchParameter,
+    openNow: true,
```

Comment:
are you assuming that they are always open? does openNow ever get set to false?

Reply:
by setting openNow to true the search query will return businesses that are currently open! Mostly included to weed out permanently closed businesses, but I'm open to adjusting this later if you'd prefer

Reply:
cool, I didn't realize the maps API had this built in functionality!